### ISSUE TITLE
physlock fails to start when run from pseudo terminal devices

### ISSUE THREAD

**Issue by bitcrush** [muennich/physlock#25]
physlock (git-20150909) fails to start when run from pseudo terminal devices with `error: could not identify active user`. It works as expected when started from a tty. This issue did not occur in older versions before [f02467f](https://github.com/muennich/physlock/commit/f02467f183d3de1e789c8187d7c859fcda78a545).  
Here's my utmp file just for reference:
```
last -f /var/run/utmp
-------------------------------------------------------------------------
rcn      tty2                          Fri Sep 18 00:58   still logged in
rcn      pts/4        tmux(1808).%3    Thu Sep 17 21:41   still logged in
rcn      pts/2        tmux(1808).%1    Thu Sep 17 21:41   still logged in
rcn      pts/0        tmux(1808).%0    Thu Sep 17 21:41   still logged in
reboot   system boot  4.1.7-1-ck       Thu Sep 17 21:35   still running

utmp begins Thu Sep 17 21:35:11 2015

```

**Comment by muennich** [comment]
What distro are you running? What init system are you using? On which tty is your X11 session running? What is the output of `ls -l /dev/tty*`?

**Comment by bitcrush** [comment]
I'm running systemd on Arch Linux. Output of `ls -l /dev/tty*` is [here](https://gist.github.com/bitcrush/46152fa6c11f27978f19). X11 is started via the login manager service. When I stop the service and start X11 directly from a tty, physlock can be started from pts. Is there a way to make physlock work with a login manager?

**Comment by muennich** [comment]
Which display/login manager are you running?

**Comment by bitcrush** [comment]
Slim on the desktop pc and lxdm on the notebook. The setup on both machines is otherwise almost identical. The issue occurs on both systems.

**Comment by muennich** [comment]
Both of these display managers do not write utmp/wtmp entries. There's a section on the Arch wiki page for slim on how to fix this: https://wiki.archlinux.org/index.php/SLiM#Login_information_with_SLiM
However, these commands need to be changed, because physlock looks for tty lines not display lines:

    sessionstart_cmd /usr/bin/sessreg -a %user
    sessionstop_cmd /usr/bin/sessreg -d %user

You also have to make sure, that sessreg runs with root rights.

On what tty is your X session running? tty2? The number typically matches the Ctrl-Alt-FX combo used to switch to it.

**Comment by bitcrush** [comment]
My X session runs on tty7. With your instructions I managed to get it to work. Thank you very much for your help.  
For anyone trying to get physlock to work with SLiM, here're some considerations:
* When using the `default_user` setting in /etc/slim.conf make sure the respective line gets parsed before `sessionstart_cmd` and `sessionstop_cmd`. Otherwise the `%user` variable will be empty and `/usr/bin/sessreg` will fail.
* I had to specify the tty my Xsession runs on directly using the `-l` parameter with `sessreg`, because autodetecting the tty failed with: `ttyname: Inappropriate ioctl for device`.

So the relevant lines in my /etc/slim.conf look like this:
```
default_user        rcn
xserver_arguments   -nolisten tcp vt07
sessionstart_cmd    /usr/bin/sessreg -a -l tty7 %user
sessionstop_cmd     /usr/bin/sessreg -d -l tty7 %user
```
